sdk/src/transaction/sanitized.rs

```diff
@@ -220,10 +219,22 @@ impl SanitizedTransaction {
                 let e = verify_eth_addresses(
                     data,
                     &instruction_datas,
-                    libsecp256k1_0_5_upgrade_enabled,
-                    libsecp256k1_fail_on_bad_count,
+                    feature_set.is_active(&feature_set::libsecp256k1_0_5_upgrade_enabled::id()),
+                    feature_set.is_active(&feature_set::libsecp256k1_fail_on_bad_count::id()),
                 );
                 e.map_err(|_| TransactionError::InvalidAccountIndex)?;
+            } else if feature_set.is_active(&feature_set::ed25519_program_enabled::id())
```

Comment:
I think it makes sense to check `crate::ed25519_program::check_id(program_id)` first.  This check will be performed for every single instruction that is ever executed, and short-circuiting the `&&` on the pubkey equality check before the `feature_set.is_active()` check should be faster

Reply:
Yes, good point.

There is something to be said for the feature set to be reduced down to a bit set rather than comparing 32 bytes every time. Having said that check_id compiles down to:
```
_ZN10solana_sdk13native_loader8check_id17hee92c47a82a05162E:
        .cfi_startproc
        movdqu  (%rdi), %xmm0
        movdqu  16(%rdi), %xmm1
        pcmpeqb .LCPI659_0(%rip), %xmm1
        pcmpeqb .LCPI659_1(%rip), %xmm0
        pand    %xmm1, %xmm0
        pmovmskb        %xmm0, %eax
        cmpl    $65535, %eax
        sete    %al
        retq
.Lfunc_end659:
```
So should be pretty quick.

Reply:
Fixed

Reply:
> There is something to be said for the feature set to be reduced down to a bit set rather than comparing 32 bytes every time

Definitely.  Optimising `FeatureSet::is_active()` has been on my wish list for a while.  Thanks for the assembly :)